controllers/cluster_controller.go

```diff
@@ -157,14 +160,14 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *ClusterReconciler) CloneTemplate(ctx context.Context, kcp *capikcp.KubeadmControlPlane) (*unstructured.Unstructured, error) {
-	infrastructureMachineTemplate, err := external.Get(ctx, r.Client, &kcp.Spec.InfrastructureTemplate, kcp.Namespace)
+func (r *ClusterReconciler) CloneTemplate(ctx context.Context, templateReference *corev1.ObjectReference, ownerNamespace, ownerName string) (*unstructured.Unstructured, error) {
```

Comment:
I changed the signature so I can use the function both in
- creating new AzureMachineTemplate for Control Plane
- creating new AzureMachineTemplate for MachineDeployment